File: TransCoder_tokenized_test_set_functions/cpp/PAIR_WITH_GIVEN_PRODUCT_SET_1_FIND_IF_ANY_PAIR_EXISTS_1.cpp
bool isProduct ( int arr [ ], int n, int x ) {
  if ( n < 2 ) return false;
  unordered_set < int > s;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( arr [ i ] == 0 ) {
      if ( x == 0 ) return true;
      else continue;
    }
    if ( x % arr [ i ] == 0 ) {
      if ( s . find ( x / arr [ i ] ) != s . end ( ) ) return true;
      s . insert ( arr [ i ] );
    }
  }
  return false;
}

File: results/java_codex/PAIR_WITH_GIVEN_PRODUCT_SET_1_FIND_IF_ANY_PAIR_EXISTS_1.java
bool isProduct(int arr[], int n, int x) {
        unordered_set<int> hset;
        if (n < 2) return false;
        for (int i = 0; i < n; i++) {
            if (arr[i] == 0) {
                if (x == 0) return true;
                else continue;
            }
            if (x % arr[i] == 0) {
                if (hset.find(x / arr[i]) != hset.end()) return true;
                hset.insert(arr[i]);
            }
        }
        return false;
    }